modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/osb-site-initializer-partner-portal-test/src/testFunctional/macros/PRMMDFRequest.macro

```diff
@@ -147,6 +147,51 @@ definition {
 		}
 	}
 
+	@summary = "Fill budget field"
+	macro fillRequestAmountField {
+		if (!(isSet(activityIndex))) {
+			var activityIndex = 0;
+		}
+
+		if (!(isSet(budgetIndex))) {
+			var budgetIndex = 0;
+		}
+
+		for (var bugdetValues : list ${budgetList}) {
+			var expenseOption = StringUtil.extractFirst(${bugdetValues}, "-");
+			var bugdetValue = StringUtil.extractLast(${bugdetValues}, "-");
+
+			Click(
+				key_buttonText = "Add Expense",
+				locator1 = "PRMMDFRequest#ADD_BUTTON");
+
+			Select(
+				key_fieldName = "activities[${activityIndex}].budgets[${budgetIndex}].expense",
+				locator1 = "PRMMDFRequest#SELECT",
+				value1 = ${expenseOption});
+
+			Type(
+				key_fieldName = "activities[${activityIndex}].budgets[${budgetIndex}].cost",
+				locator1 = "PRMMDFRequest#TEXT_FIELD",
+				value1 = ${bugdetValue});
+
+			var budgetIndex = ${budgetIndex} + 1;
+		}
+
+	}
+
+	@summary = "verify if the requested amount is equal to the mdf values"
+	macro verifyRequestedAmount {
+		if (!(isSet(activityIndex))) {
+			var activityIndex = 0;
+		}
+
+		AssertTextEquals(
+			key_fieldName = "activities[${activityIndex}].mdfRequestAmount",
+			locator1 = "PRMMDFRequest#TEXT_FIELD",
+			value1 = ${discountedValue});
```

Comment:
Hi @mateussandes23, this macro should calculate the Total MDF Requested Amount. The calculation must be done by dividing the budget value by 2.

After this, you should assert that the value on the field is equals to your result.